src/rust/zh.dmzj/src/lib.rs

```diff
@@ -0,0 +1,454 @@
+				title: it.get("comic_name").as_string()?.read(),
+				author: it.get("comic_author").as_string()?.read(),
+				artist: String::new(),
+				description: String::new(),
+				url: String::new(),
+				categories: Vec::new(),
+				status: MangaStatus::Unknown,
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Rtl,
+			});
+		}
+	} else {
+		let mut filters_query: String = String::from("0");
+		if !filters_list.is_empty() {
+			for i in filters_list {
+				filters_query.push_str(&helper::i32_to_string(i));
+				filters_query.push('-');
+			}
+			filters_query.pop();
+		}
+
+		let url = format!(
+			"{}/classify/{}/{}/{}.json",
+			V3_API_URL,
+			filters_query,
+			helper::i32_to_string(sort),
+			helper::i32_to_string(page)
+		);
+
+		let data = Request::new(&url, HttpMethod::Get).json().as_array()?;
+
+		for it in data {
+			let it = it.as_object()?;
+			manga_arr.push(Manga {
+				id: helper::i32_to_string(it.get("id").as_int()? as i32),
+				cover: it.get("cover").as_string()?.read(),
+				title: it.get("title").as_string()?.read(),
+				author: it.get("authors").as_string()?.read(),
+				artist: String::new(),
+				description: String::new(),
+				url: String::new(),
+				categories: it
+					.get("types")
+					.as_string()?
+					.read()
+					.split('/')
+					.collect::<Vec<_>>()
+					.iter()
+					.map(|s| String::from(s.deref()))
+					.collect(),
+				status: match it.get("status").as_string()?.read().as_str() {
+					"连载中" => MangaStatus::Ongoing,
+					"已完结" => MangaStatus::Completed,
+					_ => MangaStatus::Unknown,
+				},
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Rtl,
+			});
+		}
+	}
+
+	let len = manga_arr.len();
+	Ok(MangaPageResult {
+		manga: manga_arr,
+		has_more: if is_keyword { false } else { len != 0 },
+	})
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+	let url = format!(
+		"{}/comic/detail/{}?channel=android&version=3.0.0&timestamp={}",
+		V4_API_URL,
+		&id,
+		aidoku::std::current_date() as i64
+	);
+
+	let pb = helper::decode(&helper::get(url).string());
+	if pb.errno == 0 {
+		let pb_data = pb.data.unwrap();
+		return Ok(Manga {
+			id: id.clone(),
+			cover: pb_data.cover,
+			title: pb_data.title,
+			author: pb_data
+				.authors
+				.iter()
+				.map(|s| s.tag_name.clone())
+				.collect::<Vec<String>>()
+				.join("/"),
+			artist: String::new(),
+			description: pb_data.description,
+			url: format!("{}/info/{}.html", BASE_URL, id),
+			categories: pb_data.types.iter().map(|s| s.tag_name.clone()).collect(),
+			status: match pb_data.status[0].tag_name.as_str() {
+				"连载中" => MangaStatus::Ongoing,
+				"已完结" => MangaStatus::Completed,
+				_ => MangaStatus::Unknown,
+			},
+			nsfw: MangaContentRating::Safe,
+			viewer: match pb_data.direction {
+				0 => MangaViewer::Rtl, // Maybe? Can't find evidence.
+				1 => MangaViewer::Ltr,
+				2 => MangaViewer::Scroll,
+				_ => MangaViewer::Default,
```

Comment:
Seems to always just be 1

Reply:
https://m.dmzj.com/info/21231.html
This one gets 2.

Reply:
Alright, I would probably still have the default be rtl personally but whatever you think is best.